mitosheet/mitosheet/step_performers/pivot.py

```diff
@@ -155,4 +179,60 @@ def get_new_pivot_df_name(prev_state: State, sheet_index: int) -> str:
     while curr_df_name in prev_state.df_names:
         curr_df_name = f'{new_df_name_original}_{multiple_sheet_indicator}'
         multiple_sheet_indicator += 1
-    return curr_df_name
\ No newline at end of file
+    return curr_df_name
+
+
+def get_optional_code_to_replay_on_pivot(previous_steps: List[StepType], params: Dict[str, Any]) -> Tuple[List[str], List[str]]:
+    """
+    This function is a utility used to replay edits on top of pivot tables - specifically, 
+    it finds the code to try and execute on top of the a pivot table that is being edited.    
+    """
+    # If this is a pivot step, that has a destination sheet index, then we're going to 
+    # go look for the original step that created this pivot
+    if params.get('destination_sheet_index') is None:
+        return [], []
+
+    destination_sheet_index = params['destination_sheet_index']
+
+    optional_code: Tuple[List[str], List[str]] = ([], [])
+
+    # First, find the index of the step we have to start looking from
+    # for steps that edit the pivot table
+    starting_index = None
+    for index, step in enumerate(reversed(previous_steps)):
+        # Case 1: we find the last step that edited this pivot, and get it's extra code 
+        # that successfully executed, as we want to start by replaying this
+        if step.step_type == 'pivot' and 'destination_sheet_index' in step.params and step.params['destination_sheet_index'] == destination_sheet_index:
+            optional_code = step.execution_data.get('optional_code_that_successfully_executed', ([], []))
```

Comment:
There's no reason to do this in the first section of this function. Let's move it down to the next section

`previous_steps = previous_steps[starting_index + 1:]` -> `previous_steps = previous_steps[starting_index:]`

Reply:
I don't understand this comment.